modules/end-to-end-test/src/test/groovy/org/hawkular/inventory/rest/test/InventoryITest.groovy

```diff
@@ -0,0 +1,261 @@
+
+        /* Create a metric type */
+        response = postDeletable(path: "/hawkular/inventory/metricTypes", body: [id : responseTimeMTypeId, unit : "MILLI_SECOND"])
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/metricTypes/$responseTimeMTypeId", response.headers.Location)
+
+        /* link pingableHostRTypeId with responseTimeMTypeId */
+        response = postDeletable(path: "/hawkular/inventory/resourceTypes/$pingableHostRTypeId/metricTypes",
+                body : [id : responseTimeMTypeId])
+        assertEquals(204, response.status)
+
+        /* add a metric */
+        response = postDeletable(path: "/hawkular/inventory/$environmentId/metrics",
+                body: [ id : responseTimeMetricId, metricTypeId : responseTimeMTypeId ]);
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/$environmentId/metrics/$responseTimeMetricId", response.headers.Location)
+
+
+        /* add a resource */
+        response = postDeletable(path: "/hawkular/inventory/$environmentId/resources",
+            body: [ id : host1ResourceId, resourceTypeId: pingableHostRTypeId ])
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/$environmentId/resources/$host1ResourceId", response.headers.Location)
+
+        /* add another resource */
+        response = postDeletable(path: "/hawkular/inventory/$environmentId/resources",
+            body: [ id : host2ResourceId, resourceTypeId: pingableHostRTypeId ])
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/$environmentId/resources/$host2ResourceId", response.headers.Location)
+
+        /* link the metric to resource */
+        response = client.post(path: "/hawkular/inventory/$environmentId/resources/$host1ResourceId/metrics",
+            body: [ responseTimeMetricId ]);
+        assertEquals(204, response.status)
+
+        /* add a feed */
+        response = postDeletable(path: "/hawkular/inventory/$environmentId/feeds", body: [ id : feedId ])
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/$environmentId/feeds/$feedId", response.headers.Location)
+
+    }
+
+    @AfterClass
+    static void deleteEverything() {
+        /* the following would delete all data of the present user. We cannot do that as long as we do not have
+         * a dedicated user for running this very single test class. */
+        // def response = client.delete(path : "/hawkular/inventory/tenant")
+        // assertEquals(204, response.status)
+
+        /* Let's delete the entities one after another in the inverted order as we created them */
+        for (int i = pathsToDelete.size() - 1; i >= 0; i--) {
+            String path = pathsToDelete.get(i);
+            def response = client.delete(path : path)
+            assertEquals(204, response.status)
+        }
+
```

Comment:
I tried to add that, but looks like a GET handler is not there for all entities that I am deleting:
```
[DEBUG] wire - >> "DELETE /hawkular/inventory/resourceTypes/pingable-host-adc60b04-9eb0-4afa-804a-5fde73dab71b/metricTypes/response-time-3528d3aa-a311-4646-8958-adb2b686f1f9 HTTP/1.1[\r][\n]"
[DEBUG] wire - >> "Accept: application/json, application/javascript, text/javascript[\r][\n]"
[DEBUG] wire - >> "Authorization: Basic amRvZTpwYXNzd29yZA==[\r][\n]"
[DEBUG] wire - >> "Host: 127.0.0.1:8080[\r][\n]"
[DEBUG] wire - >> "Connection: Keep-Alive[\r][\n]"
[DEBUG] wire - >> "[\r][\n]"
[0m18:59:42,176 INFO  [org.apache.cassandra.db.Memtable] (MemtableFlushWriter:1) Completed flushing /home/ppalaga/git/hawkular/modules/end-to-end-test/target/wildfly-run/wildfly-8.2.0.Final/standalone/data/hawkular-metrics/data/system/schema_columnfamilies-45f5b36024bc3f83a3631034ea4fa697/system-schema_columnfamilies-ka-10-Data.db (949 bytes) for commitlog position ReplayPosition(segmentId=1434387575475, position=136310)
[0m[0m18:59:42,176 INFO  [org.apache.cassandra.db.ColumnFamilyStore] (MigrationStage:1) Enqueuing flush of schema_columns: 4006 (0%) on-heap, 0 (0%) off-heap
[0m[0m18:59:42,176 INFO  [org.apache.cassandra.db.Memtable] (MemtableFlushWriter:2) Writing Memtable-schema_columns@1799862981(757 serialized bytes, 14 ops, 0%/0% of on/off-heap limit)
[0m[DEBUG] wire - << "HTTP/1.1 204 No Content[\r][\n]"
[DEBUG] wire - << "Expires: 0[\r][\n]"
[DEBUG] wire - << "Connection: keep-alive[\r][\n]"
[DEBUG] wire - << "Cache-Control: no-cache, no-store, must-revalidate[\r][\n]"
[DEBUG] wire - << "X-Powered-By: Undertow/1[\r][\n]"
[DEBUG] wire - << "Server: WildFly/8[\r][\n]"
[DEBUG] wire - << "Pragma: no-cache[\r][\n]"
[DEBUG] wire - << "Content-Length: 0[\r][\n]"
[DEBUG] wire - << "Date: Mon, 15 Jun 2015 16:59:42 GMT[\r][\n]"
[DEBUG] wire - << "[\r][\n]"
[DEBUG] wire - >> "GET /hawkular/inventory/resourceTypes/pingable-host-adc60b04-9eb0-4afa-804a-5fde73dab71b/metricTypes/response-time-3528d3aa-a311-4646-8958-adb2b686f1f9 HTTP/1.1[\r][\n]"
[DEBUG] wire - >> "Accept: application/json, application/javascript, text/javascript[\r][\n]"
[DEBUG] wire - >> "Authorization: Basic amRvZTpwYXNzd29yZA==[\r][\n]"
[DEBUG] wire - >> "Host: 127.0.0.1:8080[\r][\n]"
[DEBUG] wire - >> "Connection: Keep-Alive[\r][\n]"
[DEBUG] wire - >> "[\r][\n]"
[33m18:59:42,253 WARN  [org.hawkular.inventory.rest] (default task-77) HAWKINV002001: Something bad has happened: javax.ws.rs.NotAllowedException: No resource method found for GET, return 405 with Allow header
	at org.jboss.resteasy.core.registry.SegmentNode.match(SegmentNode.java:375) [resteasy-jaxrs-3.0.10.Final.jar:]
	at org.jboss.resteasy.core.registry.SegmentNode.match(SegmentNode.java:114) [resteasy-jaxrs-3.0.10.Final.jar:]
	at org.jboss.resteasy.core.registry.RootNode.match(RootNode.java:43) [resteasy-jaxrs-3.0.10.Final.jar:]
	at org.jboss.resteasy.core.registry.RootClassNode.match(RootClassNode.java:48) [resteasy-jaxrs-3.0.10.Final.jar:]
	at org.jboss.resteasy.core.ResourceMethodRegistry.getResourceInvoker(ResourceMethodRegistry.java:444) [resteasy-jaxrs-3.0.10.Final.jar:]
	at org.jboss.resteasy.core.SynchronousDispatcher.getInvoker(SynchronousDispatcher.java:234) [resteasy-jaxrs-3.0.10.Final.jar:]
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:171) [resteasy-jaxrs-3.0.10.Final.jar:]
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:220) [resteasy-jaxrs-3.0.10.Final.jar:]
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56) [resteasy-jaxrs-3.0.10.Final.jar:]
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51) [resteasy-jaxrs-3.0.10.Final.jar:]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [jboss-servlet-api_3.1_spec-1.0.0.Final.jar:1.0.0.Final]
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:61) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at org.keycloak.adapters.undertow.UndertowAuthenticatedActionsHandler.handleRequest(UndertowAuthenticatedActionsHandler.java:66) [keycloak-undertow-adapter-1.2.0.Final.jar:1.2.0.Final]
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:56) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.server.handlers.DisableCacheHandler.handleRequest(DisableCacheHandler.java:33) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.security.handlers.AuthenticationConstraintHandler.handleRequest(AuthenticationConstraintHandler.java:51) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:45) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:63) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.security.ServletSecurityConstraintHandler.handleRequest(ServletSecurityConstraintHandler.java:56) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:58) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:70) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.security.handlers.SecurityInitialHandler.handleRequest(SecurityInitialHandler.java:76) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at org.keycloak.adapters.undertow.ServletPreAuthActionsHandler.handleRequest(ServletPreAuthActionsHandler.java:69) [keycloak-undertow-adapter-1.2.0.Final.jar:1.2.0.Final]
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:261) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:247) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:76) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:166) [undertow-servlet-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:197) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:759) [undertow-core-1.1.0.Final.jar:1.1.0.Final]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [rt.jar:1.8.0_45]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [rt.jar:1.8.0_45]
	at java.lang.Thread.run(Thread.java:745) [rt.jar:1.8.0_45]

```